src/test/java/teammates/test/cases/testdriver/HtmlHelperTest.java

```diff
@@ -135,4 +135,20 @@ public void testConvertToStandardHtmlPart() throws Exception {
         
         //other cases are tested in testComparison
     }
+    
+    @Test
+    public void testReplacement() throws IOException {
```

Comment:
Not really, because there are only 2 such processes being tested and they're targeting 2 different files (actual and expected). Also, there's not much point in testing after each replace. Let's say we test after only `HtmlHelper.injectTestProperties`. What are we comparing against? It won't pass the `HtmlHelper.assertSameHtml` because the other one is not yet processed, and strictly only after both the actual and expected are processed then the test will pass. We can check for (non-)presence of pattern/placeholder, but exhausting every single of the existing pattern/placeholder will be cumbersome. Combining both processes saves us all these hassles with no real drawback.
I don't think the comment is necessary either. I will revoke my previous opinion on "`GodModeTest` also serves the purpose of `HtmlHelperTest` mentioned", because the usage in `GodModeTest` is not _a test for_, but _an application of_ the HTML verification. If `GodModeTest` is a test for such, then the same can be said for _every single HTML verification_ in existence.